### ISSUE TITLE
Error "reached CPU time limit"

### ISSUE THREAD

**Issue by ValentineObrezanenko** [nf-core/rnasplice#158]
### Description of the bug

```
In DESeqDataSet(rse, design, ignoreRank = TRUE) :
    some variables in design formula are characters, converting to factors
  Error in fn(par, ...) : reached CPU time limit
  Calls: lapply ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
  Execution halted
```
Good afternoon! I put a small sample (34 samples) into processing - everything went well, without errors. I set a sample for 130 samples, constantly crashes at this place with such an error. Tell me, what could be the problem? I have increased the memory and time limits. Thank you.

### Command used and terminal output

In DESeqDataSet(rse, design, ignoreRank = TRUE) :
    some variables in design formula are characters, converting to factors
  Error in fn(par, ...) : reached CPU time limit
  Calls: lapply ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
  Execution halted

**Comment by piplus2** [comment]
Yes, that's probably R complaining because the process was taking too long. It may be either a limit of the computational environment, or something else that I need to investigate. I'm moving to enhancement as it is not a bug.

**Comment by bobems1215** [comment]
Hey I also had this error, and I did some digging (with the help of AI, so take it with a grain of salt). My problem started because the NFCORE_RNASPLICE:RNASPLICE:DEXSEQ_DEU:DEXSEQ_EXON process was failing due to the process timing out. I first fixed this with a custom config file that looked like this to increase the runtime of high processes 

`process {
    withLabel:process_high { 
        cpus = 15
        memory = 60.GB
        time = 192.h
    }
}
`
I knew that BiocParallel was loaded, so I assumed that it was parallelizing, which is why I also upped the CPUs to 15. This worked for a smaller dataset of 50 samples, but when I increased this to a full run of 150 samples, I got the CPU time limit error mentioned above. After digging into the code, I found two things.

First, in the Singularity container that rnasplice was running in, there was a bptimeout: 2592000, which is 30 days. When split between 15 processors (requested in my custom config file) was 48 hours, which was how long this would run before it was killed with the CPU time limit error. 

Second, when I tried to find any mention of MulticoreParam, BPPARAM, or bpparam to adjust the timeout settings to the integer.max (around ~600,000 hours), I couldn't find any mention of it in the run_dexseq_exon.R script. 

So what I concluded was that when I requested 15 processors for this specific command, the container's BLAS library was spawning 15 threads, and this was burning through the CPU time, but it was only ever using one single process (since BiocParrallel was loaded but never configured).

This leads to my question for the team: is this logic correct? Also, is there a way to parallelize NFCORE_RNASPLICE:RNASPLICE:DEXSEQ_DEU:DEXSEQ_EXON?

**Comment by piplus2** [comment]
Hi, the `dev` version now parallelises dexseq. Sometimes it may crash because of too much memory required, but is retries 3 times, and in my tests I've seed that at the second attempt it goes through.

I'm closing the issue. If you still have it, please reopen it.
